### DISCUSSION TITLE
How to add a button in UI to toggle "autoload=yes/no",  which is defined in script-opts/uosc.conf?

### DISCUSSION THREAD

**Discussion by rsyqvthv** [tomasklaen/uosc#1004] [Q&A] [answered]
Most of the time, I am satisfied with `autoload=no`, 
but sometimes, I would like to use `autoload=yes` when inspecting many video files.

I am a newbie at mpv scripting, I tried the following in `input.conf`, which doesn't work:

`F2 script-message cycle-cmd "script-message uosc/autoload yes" "script-message uosc/autoload no" 

Any advice?

**Comment by tomasklaen** [comment, marked as answer]
You can can use mpv's script-opts command. In your case that would be:

```
cycle-values script-opts uosc-autoload=yes uosc-autoload=no
```

I also just pushed a commit that adds support for using uosc options as cycle/toggle control property by setting its owner to `uosc`. Example:

```
controls=toggle:hdr_auto:autoload@uosc
```

**Reply by rsyqvthv** [reply]
Thank you! I am using the `cycle-values` option in `uosc.conf`, which is working great:

```
command:rotate_left:cycle-values script-opts uosc-autoload=yes uosc-autoload=no?autoload
```

It's nearly perfect, but this button doesn't show the status. I like that it has a solid white background when toggled on and cancels the background when toggled off, similar to how the "loop button" does.

How can I achieve this toggle button style?

**Reply by tomasklaen** [reply]
That's exactly what my second example in the post above does. And to reiterate: this is only on main branch atm, not released yet, so if you want to use it, you have to update your uosc manually with files from the repo (`src/uosc`)

**Reply by rsyqvthv** [reply]
Thank you very much, I am not quite get how the `controls` work, I will try it later for the toggle style, you really help me a lot, thank you.
